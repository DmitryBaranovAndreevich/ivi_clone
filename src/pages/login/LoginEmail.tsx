import { FormEvent, useEffect, useState } from 'react';
import Input from '../../components/UI/Input/Input';
import RedButton from '../../components/UI/redButton/RedButton';
import Line from '../../components/login/Line/Line';
import LoginHeader from '../../components/login/LoginHeader/LoginHeader';
import LoginLayout from '../../components/login/LoginLayout/LoginLayout';
import LoginMessage from '../../components/login/LoginMessage/LoginMessage';
import useForm from '../../hooks/useForm';
import styles from './login.module.scss';
import * as Yup from 'yup';
import { useNavigate } from 'react-router-dom';
import ErrorMessage from '../../components/login/ErrorMessage/ErrorMessage';
import { useAppDispatch, useAppSelector } from '../../hooks/redux';
import { userLoginSlice } from '../../store/reducers/UserLoginSlice';

const LoginEmail = () => {
   const { setEmail } = userLoginSlice.actions;
  const { email } = useAppSelector((state) => state.userLoginReduser);
  const dispatch = useAppDispatch();
  const { value, handleChange } = useForm({ email });
  const [error, setError] = useState('');
  const navigate = useNavigate();

  const shema = Yup.object().shape({
    email: Yup.string()
      .email('Введены не корректные данные')
      .required('Поле обязательно для заполнения'),
  });

  useEffect(() => {
    setError('');
  }, [value]);

  const handleClick = (e: FormEvent) => {
    e.preventDefault();
    shema
      .validate(value)
      .then((res) => {
        dispatch(setEmail(res.email));
        navigate('/profile/password');
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  return (
    <LoginLayout>
      <LoginHeader title={'Вход или регистрация'} />
      <Line persent={'10%'} />
      <form className={styles.wrapper}>
        <LoginMessage>
          <span className={styles.span}>Войдите или зарегистрируйтесь</span> чтобы пользоваться
          сервисом на любом устройстве
        </LoginMessage>
        <Input
          placeholder={'Через email или телефон'}
          name={'email'}
          value={value.email}
          onChange={handleChange}
        />
        <RedButton
          addingClass={styles.button}
          text={'Продолжить'}
          type={'submit'}
          onClick={handleClick}
        />
        {error && <ErrorMessage error={error} />}
      </form>
    </LoginLayout>
  );
};

export default LoginEmail;
