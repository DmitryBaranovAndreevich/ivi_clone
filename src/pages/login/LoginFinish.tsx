import { useEffect } from 'react';
import Line from '../../components/login/Line/Line';
import LoginHeader from '../../components/login/LoginHeader/LoginHeader';
import LoginLayout from '../../components/login/LoginLayout/LoginLayout';
import LoginMessage from '../../components/login/LoginMessage/LoginMessage';
import { useAppSelector } from '../../hooks/redux';
import styles from './login.module.scss';
import { useNavigate } from 'react-router-dom';

const LoginFinish = () => {
  const navigate = useNavigate();
  const { email, password } = useAppSelector((state) => state.userLoginReduser);
  const redirect = () => {
    navigate('/', { replace: true });
  };
  useEffect(() => {
    const timer = setTimeout(redirect, 4500);
    return () => {
      clearTimeout(timer);
    };
  });

  return (
    <LoginLayout>
      <LoginHeader title={'Здравствуйте'} />
      <Line persent={'100%'} />
      <LoginMessage>Войдите</LoginMessage>
      <div className={`${styles.responce}`}>
        <LoginMessage response>{email}</LoginMessage>
      </div>
      <LoginMessage>Введите пароль, чтобы войти</LoginMessage>
      <div className={`${styles.responce} ${styles.animation} ${styles.animationTime1}`}>
        <LoginMessage response>{'⚹'.repeat(password.length)}</LoginMessage>
      </div>
      <div className={`${styles.div} ${styles.animation} ${styles.animationTime2}`}>
        <LoginMessage success>Вы успешно авторизировались</LoginMessage>
      </div>
    </LoginLayout>
  );
};

export default LoginFinish;
