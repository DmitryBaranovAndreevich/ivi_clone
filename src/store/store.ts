import reviewReducer from './reducers/ReviewSlice';
import moviesSortReduser from './reducers/MoviesSort';
import { appApi } from './api/appApi';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import userLoginReduser from './reducers/UserLoginSlice';
import appReducer from './reducers/App';
import moviesFilterReduser from './reducers/MoviesFilter';
import { filmApi } from './api/filmApi';
import { personApi } from './api/personApi';
import { reviewApi } from './api/reviewApi';

const rootReducer = combineReducers({
  appReducer,
  userLoginReduser,
  moviesSortReduser,
  reviewReducer,
  moviesFilterReduser,
  [appApi.reducerPath]: appApi.reducer,
  [filmApi.reducerPath]: filmApi.reducer,
  [personApi.reducerPath]: personApi.reducer,
  [reviewApi.reducerPath]: reviewApi.reducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat([appApi.middleware, filmApi.middleware, personApi.middleware]),
  });
};

export type TRootState = ReturnType<typeof rootReducer>;
export type TAppStore = ReturnType<typeof setupStore>;
export type TAppDispatch = TAppStore['dispatch'];
