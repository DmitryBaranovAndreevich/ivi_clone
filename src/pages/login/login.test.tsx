import { Provider } from 'react-redux';
import { store } from '../../store/mainStore';
import renderer from 'react-test-renderer';
import LoginEmail from './LoginEmail';
import LoginFinish from './LoginFinish';
import * as router from 'react-router';
import { MemoryRouter } from 'react-router-dom';

describe('Тестим страницу входа', () => {
  const mockUseLocationValue = {
    key: 'key',
    pathname: '/testroute',
    search: '',
    hash: '',
    state: null,
  };
  const navigate = jest.fn();

  jest.mock('react-i18next', () => ({
    useTranslation: jest.fn(),
  }));

  beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    jest.spyOn(router, 'useLocation').mockImplementation(() => mockUseLocationValue);
  });
  it('Страница loginEmail рендерится без ошибок', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LoginEmail />
        </MemoryRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Страница loginFinish рендерится без ошибок', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LoginFinish />
        </MemoryRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
