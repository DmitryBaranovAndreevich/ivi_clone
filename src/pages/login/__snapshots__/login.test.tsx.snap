// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим страницу входа Страница loginEmail рендерится без ошибок 1`] = `
<div>
  <div>
    <p>
      login.enter
    </p>
    <button
      className="undefined undefined"
      onClick={[Function]}
    >
      <img
        alt="cross"
        src={{}}
      />
    </button>
  </div>
  <a
    href="/auth/email"
    onClick={[Function]}
    style={
      {
        "textDecoration": "none",
      }
    }
  >
    <p>
      login.register
    </p>
  </a>
  <div
    style={
      {
        "--size": "10%",
      }
    }
  />
  <form>
    <p
      style={{}}
    >
      <span>
        login.enter1
      </span>
       
      login.use
    </p>
    <div>
      <span>
        login.placeholder
      </span>
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      className="undefined "
      onClick={[Function]}
      type="submit"
    >
      login.continue
    </button>
  </form>
</div>
`;

exports[`Тестим страницу входа Страница loginFinish рендерится без ошибок 1`] = `
<div>
  <div>
    <p>
      login.greeting
    </p>
    <button
      className="undefined undefined"
      onClick={[Function]}
    >
      <img
        alt="cross"
        src={{}}
      />
    </button>
  </div>
  <a
    href="/auth/email"
    onClick={[Function]}
    style={
      {
        "textDecoration": "none",
      }
    }
  >
    <p>
      login.register
    </p>
  </a>
  <div
    style={
      {
        "--size": "100%",
      }
    }
  />
  <p
    style={{}}
  >
    login.enter1
  </p>
  <div
    className="undefined"
  >
    <p
      style={
        {
          "borderRadius": "10px 10px 0 10px",
        }
      }
    />
  </div>
  <p
    style={{}}
  >
    login.passEnt
  </p>
  <div
    className="undefined undefined undefined"
  >
    <p
      style={
        {
          "borderRadius": "10px 10px 0 10px",
        }
      }
    />
  </div>
  <div
    className="undefined undefined undefined"
  >
    <p
      style={
        {
          "backgroundColor": "lightgreen",
          "color": "black",
        }
      }
    >
      login.finish
    </p>
  </div>
</div>
`;
